targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC/trng_api.c

```diff
@@ -0,0 +1,57 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "fsl_rng.h"
+#include "trng_api.h"
+
+#if defined(DEVICE_TRNG)
+
+void trng_init(trng_t *obj)
+{
+}
+
+void trng_free(trng_t *obj)
+{
+}
+
+int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
+{
+    uint32_t skip;
+    uint32_t data;
+    size_t idx = 0;
+    int i;
+
+    /* Get Random data */
+    while (idx < length) {
+
+        data = RNG_GetRandomData();
+
+        for (i = 0; ((i < 4) && (idx < length)); i++) {
+            output[idx++] = (data >> (i * 8)) & 0xFF;
+        }
+
+        /* Skip next 32 random numbers for better entropy */
```

Comment:
How is that guaranteeing better entropy? Is there some known correlation between subsequent bytes returned from the RNG?

Reply:
This is done per what is recommended in the reference manual. 
_________________________________________
To constitute one 128 bit number, a 32 bit random number is read, then the next 32
numbers are read but not used. The next 32 bit number is read and used and so on. Thus
32 32-bit random numbers are skipped between two 32-bit numbers that are used.
__________________________________________

Reply:
The code doesn't implement this but instead skips 32*32 bits per 32 bits of entropy. Could you correct that, to avoid unnecessary workload in the device?

Reply:
For every 32 bit random number read, it is recommended that the next 32 32-bit random numbers are skipped. This is what the code is doing.

Reply:
I should have read that properly - apologies, my fault!